src/Functions/ztest.cpp

```diff
@@ -0,0 +1,199 @@
+#include <Common/typeid_cast.h>
+#include <Columns/ColumnNullable.h>
+#include <Columns/ColumnTuple.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/castTypeToEither.h>
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Interpreters/castColumn.h>
+#include <boost/math/distributions/normal.hpp>
+
+
+namespace DB
+{
+
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    }
+
+
+    class FunctionTwoSampleProportionsZTest : public IFunction
+    {
+    public:
+        static constexpr auto POOLED = "pooled";
+        static constexpr auto UNPOOLED  = "unpooled";
+
+        static constexpr auto name = "proportionsZTest";
+
+        static FunctionPtr create(ContextPtr)
+        {
+            return std::make_shared<FunctionTwoSampleProportionsZTest>();
+        }
+
+        String getName() const override
+        {
+            return name;
+        }
+
+        size_t getNumberOfArguments() const override { return 6; }
+        ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {5, 6}; }
+
+        bool useDefaultImplementationForNulls() const override { return false; }
+        bool useDefaultImplementationForConstants() const override { return true; }
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+        static DataTypePtr getReturnType()
+        {
+            DataTypes types
+            {
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+                std::make_shared<DataTypeNumber<Float64>>(),
+            };
+
+            Strings names
+            {
+                "z_statistic",
+                "p_value",
+                "confidence_interval_low",
+                "confidence_interval_high"
+            };
+
+            return std::make_shared<DataTypeTuple>(
+                std::move(types),
+                std::move(names)
+            );
+        }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+
+            for (size_t i = 0; i < 4; ++i)
+            {
+                if (!isNativeInteger(arguments[i]))
+                {
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "The {}th Argument of function {} must be a integer or unsigned integer.", i + 1, getName());
```

Comment:
Consider to use fmt exception formatting style.

Reply:
It's already like this